src/StateMachines/StateMachine.php

```diff
@@ -96,6 +100,9 @@ public function hasPendingTransitions()
      */
     public function transitionTo($from, $to, $customProperties = [], $responsible = null)
     {
+        $from = $this->normalizeEnumCasting($from);
+        $to = $this->normalizeEnumCasting($to);
```

Comment:
Would it be good to preserve the original enums so that they can be passed to the `before` and `after` hooks with their original types?

Reply:
About this, I'm not sure if that change will bring major benefits for the project, since it will change a good part of the structure of how the `before` and `after` hooks works today.

The point of this is that the keys of an array cannot be enum objects, so since this value cannot be used as an enum, there is no reason to preserve it within the function.

I hope this clarifies my decision-making on the case and that if you still see a way to contribute to the discussion, leave it written below, and I will look at it as soon as possible.

Thank you for your contribution so far.